### DISCUSSION TITLE
joining  a room (interface: joinOrCreate) gets slower and slower

### DISCUSSION THREAD

**Discussion by LionSummer** [colyseus/colyseus#643] [Q&A]
multiple people enter a room with the same name at the same time, and the waiting time for users to successfully join is getting longer and longer. Why is this? Is there any way to optimize or solve it?
@endel

**Comment by LionSummer** [comment]
Hi @endel  
 The current situation is that a user enters a lobby room to be assigned a battle room. When multiple users enter the lobby room at the same time, the request time will continue to be extended.
Self-test found that the number of people entering at the same time is about 200, and the delay time will be more than 6 seconds.

How to optimize this delay length?

**Comment by endel** [comment]
Hi @LionSummer, there is a concurrency checker on `joinOrCreate` in order to make sure 4 concurrent `joinOrCreate` requests join the same room, more discussion about this here: https://github.com/colyseus/colyseus/issues/466

A better solution for this must be implemented.
